app/Http/Controllers/EventController.php

```diff
@@ -16,8 +16,13 @@ class EventController extends Controller
      */
     public function index(Request $request)
     {
-        $events = Event::all();
+        $events = Event::filtered($request->get('search'));
+        if ($request->has('registered_user') || $request->has('registered_only')) {
+            $events = $events->whereRegistered($request->get('registered_user'));
+        }
+        $events = $events->get();
         $events->load('league');
```

Comment:
I recall you saying something about using `->with(...)` instead of `->load()` for this, and you're absolutely right we can do that here instead, and in fact its more efficient: 
```suggestion
        $events = Event::filtered($request->get('search'));
        if ($request->has('registered_user') || $request->has('registered_only')) {
            $events = $events->whereRegistered($request->get('registered_user'));
        }
        $events = $events->with('league')->get();
```